### PR TITLE
Telekinesis allows safe opening of traps (Fixes #1916)

### PR DESCRIPTION
**Pull request by dteviot**
When trap activated at beyond normal activation distance, assume telekinesis used and detonate trap at trapped object's location.
Also some minor code refactoring of spellcasting.
1. Corrected parameter passed to explodeSpell().
2. For loop now correctly does an early exit.
3. Removed duplicated tests.

### CODE REVIEW

File: apps/openmw/mwbase/world.hpp [hunk 1]
```diff
@@ -269,6 +269,8 @@ namespace MWBase
             virtual MWWorld::Ptr  getFacedObject() = 0;
             ///< Return pointer to the object the player is looking at, if it is within activation range
 
+            virtual float getMaxActivationDistance() = 0;
```
Comment (scrawl) [inline apps/openmw/mwbase/world.hpp]: The implementation for this function doesn't seem to be included in this pull request. Forgot to add it?

**Comment by dteviot** [comment]
> The implementation for this function doesn't seem to be included in this pull request. Forgot to add it?

No.  The implementation of the function is already present in the MWWorld::World class.
Although if you've got a better way to expose it via a MWBase::World pointer I'd like to know it.

And proof it works should be the Travis CI build passing.
